tests/route/test_static_route.py

```diff
@@ -118,33 +118,79 @@ def check_route_redistribution(duthost, prefix, ipv6, removed=False):
     # Add ipaddresses in ptf
     add_ipaddr(ptfhost, nexthop_addrs, prefix_len, nexthop_devs, ipv6=ipv6)
 
     try:
+        ip_dst = str(ipaddress.ip_network(unicode(prefix))[1])
         # Add static route
         duthost.shell("sonic-db-cli CONFIG_DB hmset 'STATIC_ROUTE|{}' nexthop {}".format(prefix, ",".join(nexthop_addrs)))
         time.sleep(5)
 
-        # Check traffic get forwarded to the nexthop
-        ip_dst = str(ipaddress.ip_network(unicode(prefix))[1])
-        generate_and_verify_traffic(duthost, ptfadapter, tbinfo, ip_dst, nexthop_devs, ipv6=ipv6)
-
-        # Check the route is advertised to the neighbors
-        check_route_redistribution(duthost, prefix, ipv6)
-
         # Config save and reload if specified
         if config_reload_test:
             duthost.shell('config save -y')
             config_reload(duthost)
-            generate_and_verify_traffic(duthost, ptfadapter, tbinfo, ip_dst, nexthop_devs, ipv6=ipv6)
-            check_route_redistribution(duthost, prefix, ipv6)
 
+        if ipv6:
+            duthost.shell("sonic-clear ndp")
+            for idx, addr in enumerate(nexthop_addrs):
+                # Add static neighbor
+                duthost.shell(
+                    "ip -6 neigh add {} lladdr {} dev {}".format(addr, ptfadapter.dataplane.get_mac(0, nexthop_devs[idx]),
+                                                              vlan_intf_devname))
+                pkt = testutils.simple_icmpv6_packet(
+                    eth_dst=duthost.facts["router_mac"],
+                    eth_src=ptfadapter.dataplane.get_mac(0, nexthop_devs[idx]),
+                    ipv6_src=addr,
+                    ipv6_dst=vlan_intf_addr,
+                    ipv6_hlim=64,
+                    icmp_code=0,
+                    icmp_type=128
+                )
+                # Send pkt to update DUT's FDB table to avoid flooding
+                testutils.send(ptfadapter, nexthop_devs[idx], pkt)
+        else:
+            duthost.shell("sonic-clear arp")
+            for idx, addr in enumerate(nexthop_addrs):
+                # Add static neighbor
+                duthost.shell(
+                    "ip neigh add {} lladdr {} dev {}".format(addr, ptfadapter.dataplane.get_mac(0, nexthop_devs[idx]),
+                                                              vlan_intf_devname))
+
+                pkt = testutils.simple_icmp_packet(
+                    eth_dst=duthost.facts["router_mac"],
+                    eth_src=ptfadapter.dataplane.get_mac(0, nexthop_devs[idx]),
+                    ip_src=addr,
+                    ip_dst=vlan_intf_addr,
+                    icmp_type=8,
+                    icmp_code=0,
+                    ip_ttl=64
+                )
+                # Send pkt to update DUT's FDB table to avoid flooding
+                testutils.send(ptfadapter, nexthop_devs[idx], pkt)
```

Comment:
Does this part does the neighbor resolution? If so, we probably need to avoid it since the automatic neighbor resolution is part of the static route feature, we need to check that. I think the current failures of the test come from two aspects: 1. the arp_update script has a period of 300 seconds, whereas the config reload waiting time is shorter than that; 2. according to https://github.com/Azure/sonic-buildimage/issues/4930 the arp_update some times fails.

Reply:
No, it is to fill the FDB entry for all interfaces on the DUT to avoid flooding unknown unicast packets. I think we may not need this changes anymore if we can wait until arp_update script is finished.